internal/controller/management/upgrade/common.go

```diff
@@ -0,0 +1,54 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const v050CompatibilityLabelKey = "kargo.akuity.com/v0.5.0-compatible"
```

Comment:
How would we delete this label in the future?

Reply:
I had been thinking we don't care if it stays.